x-pack/plugins/drilldowns/public/types.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AdvancedUiActionsActionFactoryDefinition as ActionFactoryDefinition } from '../../advanced_ui_actions/public';
+
+export interface Drilldown<
+  Config extends object = object,
+  CreationContext extends object = object,
+  ExecutionContext extends object = object
+> {
+  /**
+   * Globally unique identifier for this drilldown.
+   */
+  id: string;
+
+  /**
+   * List of places where this drilldown should be available, e.g "dashboard", "graph".
+   * If omitted, the drilldown will be shown in all places.
+   */
+  places?: string[];
```

Comment:
> Is this used anywhere now?

Yes, it is passed to action factories to let them know where they are being rendered.

![image](https://user-images.githubusercontent.com/9773803/77149275-7e4d0380-6a91-11ea-97b7-a85461038567.png)

> I don't see how this will be used to check for where the drilldown will show up.

*Graph-to-Discover* drilldown will specify in `places` field that it should be rendered only on dashboard and *APM-to-URL* will specify in `places` field that it should be rendered only in APM app.

> Dashboard Embeddable doesn't know where it is embedded.

*APM-to-URL* drilldown would work only in APM, it has to know where it is being rendered so it is rendered only in APM app, not in Dashboard, for example.

> Do we even want to restrict drilldowns to certain apps?

Ideally we would want to re-use code and make some of the drilldowns generic, but right now all below drilldowns would be separate implementations of drilldowns:

- *Dashboard-to-Dashboard drilldown*
- *Dashboard-to-Discover drilldown*
- *Dashboard-to-URL drilldown*
- *Dashboard-to-ML drilldown*
- *Dashboard-to-Logs drilldown*
- *APM-to-URL drilldown*
- *APM-to-Dashboard drilldown*
- *Graph-to-Discover drilldown*
- *Graph-to-Dashboard drilldown*
- *ML-to-URL drilldown*

I've removed `places` in this PR to avoid confusion, as this PR adds drilldonws only to Dashboard, this will display all action factories in drilldown picker, which should be fine as nobody else should be using the action factory registry for now.